### ISSUE TITLE
view_multichannel function does not behave as described

### ISSUE THREAD

**Issue by GenevieveBuckley** [napari/napari#632]
## 🐛 Bug

The mini-tutorial on `view_multichannel` @jni presents in [his blogpost](https://ilovesymposia.com/2019/10/24/introducing-napari-a-fast-n-dimensional-image-viewer-in-python/) doesn't work for the current release (0.2.3) or for napari installed from the master branch.

Also, https://github.com/napari/napari/pull/619 makes is sound as though `view_multichannel` is going away very soon. If that's really the case, maybe the blog post will need to be updated then.

## To Reproduce

1. pip install napari (either the latest release 0.2.3, or from the master branch). Check the installation has worked (using `napari.view_image()` or similar with the scikit-image camera and astronaut demo images).
2. Download the [mitosis dataset](http://fiji.sc/downloads/Spindly-GFP.zip) and extract *mitosis.tif*
3. Try `napari.view_multichannel` as described [in the blogpost](https://ilovesymposia.com/2019/10/24/introducing-napari-a-fast-n-dimensional-image-viewer-in-python/)
```python
import napari
from skimage import io

mitosis = io.imread('mitosis.tif')

viewer = napari.view_multichannel(
    mitosis, axis=2, colormap=['magenta', 'green'],
    name=['aurora-B', 'tubulin'], scale=[1, 10, 1, 1],
    contrast_limits=[[1500, 6500], [1600, 16000]],
)
```
Error: napari has no attribute view_multichannel

Ok, fine. It seems the `view_multichannel` function from `view_layers.py` is not being imported into the main napari namespace. That's ok, I'll just do this...

```python
from napari.view_layers import view_multichannel

view_multichannel(
    mitosis, axis=2, colormap=['magenta', 'green'],
    name=['aurora-B', 'tubulin'], scale=[1, 10, 1, 1],
    contrast_limits=[[1500, 6500], [1600, 16000]],
)
```
... and now I get 

```python-traceback
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-25-498a9d0d501f> in <module>
----> 1 view_multichannel(mitosis, axis=2, colormap=['magenta', 'green'], scale=[1, 10, 1, 1])

c:\users\genevieb\.conda\envs\napari\lib\site-packages\napari\view_layers.py in view_multichannel(data, axis, colormap, contrast_limits, interpolation, rendering, name, metadata, scale, translate, opacity, blending, visible, path, title, ndisplay, order)
    194         data=data,
    195         axis=-1,
--> 196         colormap=cmap,
    197         contrast_limits=clims,
    198         interpolation=interpolation,

NameError: name 'cmap' is not defined
```

Looking at the code in view_layers.py this does make sense: `cmap` and `clims` are never defined, the things being passed in are called `colormap` and `contrast_limits`. 

## Expected behavior

I expected to see a multidimensional image in the napari GUI, like the gif @jni shows.

## Environment

 - Please copy and paste the information at napari info option in help menubar here:

napari, 0.2.3
Qt, 5.13.1
NumPy, 1.17.3
SciPy, 1.3.1
VisPy, 0.6.1
scikit-image, 0.16.2

Platform: Windows-10-10.0.17763-SP0
Python:   3.7.4 (default, Aug  9 2019, 18:34:13) [MSC v.1915 64 bit (AMD64)]
Backend:  PySide2
GL version:  '4.5.0 - Build 25.20.100.6472'
MAX_TEXTURE_SIZE: 16384

 - Any other relevant information:

**Windows napari development install**
Originally, this worked great for me out of the box with just a git clone and then `pip install -e .`

Unfortunately after I made a new branch and committed two very minor changes [(here's the branch)](https://github.com/GenevieveBuckley/napari/tree/multichannel-in-main-namespace), I started seeing crashes like this:

```python
Python 3.7.4 (default, Aug  9 2019, 18:34:13) [MSC v.1915 64 bit (AMD64)]
Type 'copyright', 'credits' or 'license' for more information
IPython 7.9.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import napari

In [2]: viewer = napari.view_image()
WARNING: QWidget: Must construct a QApplication before a QWidget
```
ipython now crashes entirely any time I try to launch the GUI.

I can't recover by switching back to the (previously working) master branch. I get this crash every time.

**Ubuntu with napari**
I tried making an empty conda environment for python 3.7, then pip installing napari (v0.2.3) into it. Unfortunately I saw the the same QWidget error as above, so the installation has not worked. I'm unsure why things would work worse on Linux, since that's supposed to be an easier installation than Windows.

## Additional context
https://github.com/napari/napari/pull/617
https://github.com/napari/napari/issues/615

**Comment by GenevieveBuckley** [comment]
To be clear, I think the issue with view_multichannel should be pretty straightforward and I'm happy to put some effort into fixing that. 

But I am pretty worried about how unstable the different installations I've tried seem be. That's going to make things hugely difficult to contribute.

**Comment by sofroniewn** [comment]
Yup, that's clearly a problem, apologies, - evidently we never added a test for `view_multichannel` to `tests/test_napari.py` and so we missed that and when #615 and #617 got raised and closed the problem was them not being in the `init` and evidently added to the name space. I don't think there's actually ever been a time when `view_multichannel` has actually been run.

I think we should merge #619 and then update the blog post (possibly making the 0.2.4 release right away). Thoughts @jni?

For the following issue
```
WARNING: QWidget: Must construct a QApplication before a QWidget
```
are you launching with `IPython --gui=qt5` or running `%gui qt5` - see our [getting started tutorial](http://napari.org/tutorials/getting_started.html) for more info. That is the error message that happens if you try and launch napari before instantiating the qt gui

**Comment by jni** [comment]
Yeah, the timeline is:

- I didn't find `view_multichannel`
- I created my branch and made #615 and #617, *and* wrote the blog post
- @sofroniewn told me actually your master is out of date, which seemed legit
- I closed those and updated my master, but never re-ran that part of the code

This totally needs an update to the code/post. I might remove the `view_multichannel` use, since I have the double add_image there which should work. I've reviewed #619 and will aim to merge that soon and push out a new release, but I don't want the post to be waiting on that.

@GenevieveBuckley re: contributing, I've actually found it quite fun to be contributing to napari/alpha software, because backwards compatibility is a much smaller concern than for scikit-image! In skimage you are kinda coding with one hand strapped behind your back. But anyway, as @sofroniewn said, the long term fix is #619, so we should work with that.

**Comment by GenevieveBuckley** [comment]
> I don't think there's actually ever been a time when `view_multichannel` has actually been run.

Juan clearly managed it somehow, I wondered if he might be using another branch or something.

> are you launching with `IPython --gui=qt5` or running `%gui qt5` - see our [getting started tutorial](http://napari.org/tutorials/getting_started.html) for more info. That is the error message that happens if you try and launch napari before instantiating the qt gui

Oops, I think I only remember to do this sometimes! That's what I'm doing wrong here :) Uh, is there a way we can display a more meaningful error message (like what you just said here)? I'm sure I'm not the only person who'll do this, and I'm also sure I'm going to forget and do the same thing again!

**Comment by jni** [comment]
> Juan clearly managed it somehow, I wondered if he might be using another branch or something.

Yep, see my comment above

> Uh, is there a way we can display a more meaningful error message (like what you just said here)? I'm sure I'm not the only person who'll do this, and I'm also sure I'm going to forget and do the same thing again!

Yes, we absolutely need to do this, see #634 to track this issue.

**Comment by sofroniewn** [comment]
The `view_mulitchannel` issue has been solved by #619, where that part of the api has been deprecated in favor of 
```python
add_image(data, channel_axis=-1)
```

I will close this issue, but please let us know if this fix doesn't work for you @GenevieveBuckley.
As you can tell our main apis are still a little in flux, but they should be settling down soon!!

We're also now tracking the event loop issue in #634
